### ISSUE TITLE
Add DAT tests for Rust

### ISSUE THREAD

**Issue by wjones127** [delta-io/delta-rs#863]
# Description

Data Acceptance tests should be done for the Rust Arrow reader.

See: https://github.com/delta-incubator/dat

**Use Case**

**Related Issue(s)**

**Comment by xuky76** [comment]
@wjones127 is this still current? (If so) how would you like to see this completed?

**Comment by wjones127** [comment]
Yes, this is still current. We have these for the Python bindings, but not for Rust.

For reference, here are the scripts for the Python tests:

 * [Download script](https://github.com/delta-io/delta-rs/blob/da101273635224a703ff7e1abd435d985b879f8c/python/Makefile#L19-L26)
 * [Test script](https://github.com/delta-io/delta-rs/blob/main/python/tests/data_acceptance/test_reader.py)

So we'd want something parallel to that. This could be based on the [load operation](https://github.com/delta-io/delta-rs/blob/main/rust/src/operations/load.rs), or perhaps the DataFusion TableProvider.

It might be a bit challenging to write in Rust, since the tests need to be generated based on the contents of a directory. You can see the approach `delta-kernel-rs` took here: https://github.com/delta-incubator/delta-kernel-rs/blob/main/tests/dat_reader.rs

**Comment by xuky76** [comment]
Cool. I have put in a draft. There are some parts missing but the idea seems to be working mostly
